### PR TITLE
#gh10655 use contract term pricesystem if available when calculating Quantity in Price Unit and Line Amount

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/10655

### CODE REVIEW

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/subscription/impl/SubscriptionBL.java [hunk 1]
```diff
@@ -1018,4 +997,79 @@ public boolean isActiveTerm(@NonNull final I_C_Flatrate_Term term)
...
+		}
+
+		// priceQty is the qty do be delivered during one complete subscription term
+		final Quantity priceQty = qtyOrderedPerRun.multiply(numberOfRuns);
+
+		// qty ordered needs to be set because it will be used to compute the
+		// line's NetLineAmount in MOrderLine.beforeSave()
+		ol.setQtyOrdered(priceQty.toBigDecimal());
+
+		ol.setQtyEnteredInPriceUOM(priceQty.toBigDecimal());
```
Comment (metas-ts) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/subscription/impl/SubscriptionBL.java:1063]: i think priceQty needs to be converted to the `ol`s C_UOM (with uomConversionBL)

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/subscription/impl/SubscriptionBL.java [hunk 2]
```diff
@@ -1018,4 +997,79 @@ public boolean isActiveTerm(@NonNull final I_C_Flatrate_Term term)
 
 		return !isCancelledOrVoided;
 	}
+
+	@Override
+	public void updatePrices(
```
Comment (metas-ts) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/subscription/impl/SubscriptionBL.java:1002]: plese call it e.g. updateQtysAndPrices

**Review by metas-ts — CHANGES_REQUESTED** [review]

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/subscription/impl/SubscriptionBL.java (hunk 2 above)
Comment (adi-stefan) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/subscription/impl/SubscriptionBL.java]: Done

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/subscription/impl/SubscriptionBL.java (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/subscription/impl/SubscriptionBL.java]: Was unsure if I should throw an error if the conversion fails, or just use the `priceQty` result. Error made more sense. Feel free to disagree.

**Review by metas-ts — APPROVED** [review]

**Pull request merged**